Description: Don't consider users above UID_MAX to be system users.
 By default users above UID 60000 will not show on the login screen and this is a change from 12.04 and confusing for sysadmins.
 If you increase UID_MAX then the nobody and nfsnobody users show on the login screen.
 Instead only consider users to be below UID_MIN and blacklist the two system users above this.
Bug-Ubuntu: https://launchpad.net/bugs/1290785
Author: Robert Ancell <robert.ancell@canonical.com>
Forwarded: not-needed

diff --git a/src/user-classify.c b/src/user-classify.c
index b68c9ae..0067938 100644
--- a/src/user-classify.c
+++ b/src/user-classify.c
@@ -195,15 +195,14 @@ user_classify_parse_login_defs_field (const gchar *contents,
 }
 
 static void
-user_classify_read_login_defs (uid_t *min_uid,
-                               uid_t *max_uid)
+user_classify_read_login_defs (uid_t *min_uid)
 {
         GError *error = NULL;
         char *contents;
 
         if (!g_file_get_contents ("/etc/login.defs", &contents, NULL, &error)) {
-                g_warning ("Could not open /etc/login.defs: %s.  Falling back to default human uid range of %d to %d",
-                           error->message, (int) *min_uid, (int) *max_uid);
+                g_warning ("Could not open /etc/login.defs: %s.  Falling back to default minimum human uid of %d", 
+                           error->message, (int) *min_uid);
                 g_error_free (error);
                 return;
         }
@@ -212,25 +211,26 @@ user_classify_read_login_defs (uid_t *min_uid,
                 g_warning ("Could not find UID_MIN value in login.defs.  Using default of %d", (int) *min_uid);
         }
 
-        if (!user_classify_parse_login_defs_field (contents, "UID_MAX", max_uid)) {
-                g_warning ("Could not find UID_MIN value in login.defs.  Using default of %d", (int) *max_uid);
-        }
-
         g_free (contents);
 }
 
 static gboolean
-user_classify_is_in_human_range (uid_t uid)
+user_classify_is_in_human_range (const gchar *username,
+                                 uid_t uid)
 {
-        static uid_t min_uid = 1000, max_uid = 60000;
+        static uid_t min_uid = 1000;
         static gboolean initialised;
 
         if (!initialised) {
-                user_classify_read_login_defs (&min_uid, &max_uid);
+                user_classify_read_login_defs (&min_uid);
                 initialised = TRUE;
         }
 
-        return min_uid <= uid && uid <= max_uid;
+        /* There are a few system accounts that are outside UID_MIN - ignore them */
+        if (strcmp (username, "nobody") == 0 || strcmp (username, "nfsnobody") == 0)
+                return FALSE;
+
+        return uid >= min_uid;
 }
 #endif /* ENABLE_USER_HEURISTICS */
 
@@ -243,6 +243,6 @@ user_classify_is_human (uid_t        uid,
 #ifdef ENABLE_USER_HEURISTICS
         return !user_classify_is_excluded_by_heuristics (username, shell, password_hash);
 #else
-        return user_classify_is_in_human_range (uid);
+        return user_classify_is_in_human_range (username, uid);
 #endif
 }
